lib/modules/datasource/github-tags/index.ts

```diff
@@ -21,37 +20,21 @@ export class GithubTagsDatasource extends GithubReleasesDatasource {
     this.tagsCache = new CacheableGithubTags(this.http);
   }
 
-  @cache({
-    ttlMinutes: 120,
-    namespace: `datasource-${GithubTagsDatasource.id}`,
-    key: (registryUrl: string, githubRepo: string, tag: string) =>
-      `${registryUrl}:${githubRepo}:tag-${tag}`,
-  })
   async getTagCommit(
     registryUrl: string | undefined,
-    githubRepo: string,
+    packageName: string,
     tag: string
   ): Promise<string | null> {
-    const apiBaseUrl = getApiBaseUrl(registryUrl);
-    let digest: string | null = null;
-    try {
-      const url = `${apiBaseUrl}repos/${githubRepo}/git/refs/tags/${tag}`;
-      const res = (await this.http.getJson<TagResponse>(url)).body.object;
-      if (res.type === 'commit') {
-        digest = res.sha;
-      } else if (res.type === 'tag') {
-        digest = (await this.http.getJson<TagResponse>(res.url)).body.object
-          .sha;
-      } else {
-        logger.warn({ res }, 'Unknown git tag refs type');
-      }
-    } catch (err) {
-      logger.debug(
-        { githubRepo, err },
-        'Error getting tag commit from GitHub repo'
-      );
+    let result: string | null = null;
+    const tagReleases = await this.tagsCache.getItems({
+      packageName,
+      registryUrl,
+    });
+    const tagRelease = tagReleases.find(({ version }) => version === tag);
```

Comment:
Are we assuming that we always have a cached list of tags already? Is that the case if there's no currentValue?

Reply:
> Are we assuming that we always have a cached list of tags already?

Except the case when it has been invalidated between `getReleases` and `getDigest` calls

> Is that the case if there's no currentValue?

If I understood your question correctly, this PR affects only the case when `newValue` was provided, so we return the digest for it. When `newValue` is unknown, we still query latest commit hash of the default branch.

Reply:
Do we have to worry about the case where we have no cached releases/tags? e.g. don't we skip `getPkgReleases()` in the case where `currentValue` is null or undefined?

Reply:
Do you mean what if the cache is absent/expired at the moment of `this.tagsCache.getItems()` call? It will fetch it internally and return the fresh result, so the empty result is possible only when there is now any tags/releases in the package.

Reply:
ok, if the fallback is already done then fine by me